execution/extrinsic/runtime_extrinsic.go

```diff
@@ -246,6 +329,33 @@ func (re runtimeExtrinsic) runtimeEvent(variants sc.Sequence[sc.Option[primitive
 	)
 }
 
+func (re runtimeExtrinsic) runtimeError() primitives.MetadataType {
+	return primitives.NewMetadataTypeWithPath(metadata.TypesRuntimeError, "", sc.Sequence[sc.Str]{"node_template_runtime", "RuntimeError"}, primitives.NewMetadataTypeDefinitionVariant(
+		sc.Sequence[primitives.MetadataDefinitionVariant]{
+			primitives.NewMetadataDefinitionVariant(
+				"System",
+				sc.Sequence[primitives.MetadataTypeDefinitionField]{
+					primitives.NewMetadataTypeDefinitionField(metadata.TypesSystemErrors),
+				},
+				runtimeSystemErrIdx,
+				""),
+			primitives.NewMetadataDefinitionVariant(
+				"Balances",
+				sc.Sequence[primitives.MetadataTypeDefinitionField]{
+					primitives.NewMetadataTypeDefinitionField(metadata.TypesBalancesErrors),
+				},
+				runtimeBalancesErrIdx,
+				""),
+			primitives.NewMetadataDefinitionVariant(
+				"Grandpa",
+				sc.Sequence[primitives.MetadataTypeDefinitionField]{
+					primitives.NewMetadataTypeDefinitionField(metadata.TypesGrandpaErrors),
+				},
+				runtimeGrandpaErrIdx,
+				""),
+		}))
+}
+
```

Comment:
Can we refactor it to be the same as `runtimeEvent` and `runtimeCall`?

Reply:
runtimeCall and runtimeEvent use a sequence of CallDef and EventDef respectively to build the type. So, to be the same as them, you mean to add ErrorDef in the MetadataModule struct and do it analogically ?